x-pack/plugins/secret_service/server/lib/secret_service.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import crypto from 'crypto';
+import { SavedObjectsClient } from 'src/legacy/server/saved_objects';
+// @ts-ignore
+import { AuditLogger } from '../../../server/lib/audit_logger';
+import { buildCrypt } from './crypt_keeper';
+
+type SSEvents = 'secret_object_created' | 'secret_object_accessed';
+
+export class SecretService {
+  public readonly hideAttribute: (deets: any, secretKey: string) => any;
+  public readonly unhideAttribute: (deets: any) => any;
+  private readonly type: string;
+  private readonly auditor?: AuditLogger;
+
+  constructor(
+    savedObjectsClient: SavedObjectsClient,
+    type: string,
+    key?: Buffer,
+    auditor?: AuditLogger
+  ) {
+    this.type = type;
+    this.auditor = auditor;
+    key = key || crypto.randomBytes(128);
+    const crypt = buildCrypt({ key: key.toString('hex') });
+
+    const logEvent = (event: SSEvents, message?: string) => {
+      if (this.auditor) {
+        this.auditor.log(event, message);
```

Comment:
We'll likely want to log "data" (the third parameter to AuditLogger.log) in addition to the message, so that it shows up in a more structured format that would allow analysis.

Reply:
I plan to do some enhancements to the auditlogging in this service in a follow up PR where the secret service is scoped to a request.